roles/etcd_prometheus_exporter_cert/tasks/main.yml

```diff
@@ -0,0 +1,35 @@
+- import_tasks: gen_cert.yml
+
+- name: 'create tempfile for etcd-exporteer auth secret'
+  tempfile:
+    state: file
+    suffix: yml
+  register: kube_etcd_exporter_auth
+  run_once: True
+
+- name: 'template secret'
+  template:
+    src: etcd_exporter_secret.yml.j2
+    dest: '{{ kube_etcd_exporter_auth.path|default("<check_mode_tempfile>") }}'
+  run_once: True
+
+- name: 'Deploy etcd exporter secret'
+  kube:
+    kubectl: '{{ bin_dir }}/kubectl'
+    filename: '{{ kube_etcd_exporter_auth.path|default }}'
```

Comment:
This is not exactly the same as how the path is constructed above, which seems fishy?